lib/puppet/type/powercli_esx_syslog.rb

```diff
@@ -0,0 +1,130 @@
+Puppet::Type.newtype(:powercli_esx_syslog) do
+  desc 'This type provides Puppet with the capabilities to manage syslog servers on an esx host'
+
+  ensurable
+
+  newparam(:esx_host, namevar: true) do
+    desc 'The name of the esx host the syslog servers will be managed on'
+    validate do |value|
+      unless value.is_a?(String)
+        raise ArgumentError, "esx_host is expected to be a String, given: #{value.class.name}"
+      end
+    end
+  end
+
+  newproperty(:syslog_server) do
+    validate do |value|
+      # puppet, being puppet if it's an array, it passes each value of the array into this
+      # validate function, so we just need to check that each thing is a string
+      unless value.is_a?(String)
+        raise ArgumentError, "each syslog_server is expected to be a String, given: #{value.class.name}"
+      end
+    end
+
+    # def sort_array(a)
```

Comment:
@gsperry2011 If we're only supporting one sever, let's remove the array sorting code